assignments/prototype-refactor.js

```diff
@@ -7,3 +7,177 @@ Prototype Refactor
 2. Your goal is to refactor all of this code to use ES6 Classes. The console.log() statements should still return what is expected of them.
 
 */
+
+//////////////////////////OG CODE////////////////////////////
+/*function GameObject(type) {
+  this.createdAt = type.createdAt;
+  this.name = type.name;
+  this.dimensions = type.dimensions;
+}
+
+GameObject.prototype.destroy = function() {
+  return `${this.name} was removed from the game.`
+}*/
+
+class GameObject {
+    constructor(type) {
+        this.createdAt = type.createdAt;
+        this.name = type.name;
+        this.dimensions = type.dimensions;
+    }
+    destroy() {
+        return `${this.name} was removed from the game.`;
+      }
+  }
+//////////////////////////OG CODE////////////////////////////
+/*function CharacterStats(characterAttributes) {
+  this.healthPoints = characterAttributes.healthPoints;
+  GameObject.call(this, characterAttributes);
+  }
+CharacterStats.prototype = Object.create(GameObject.prototype);
+
+CharacterStats.prototype.takeDamage = function() {
+  return `${this.name} took damage.`
+}
+////////////////////// TRYING DEFACTORING////////////////////
+class CharacterStats extends GameObject{
```

Comment:
I know you were having trouble with destructuring prior to stand up. Did our code along help?